lib/output.js

```diff
@@ -117,15 +117,37 @@ function OutputStream(options) {
         function quote_double() {
             return '"' + str.replace(/\x22/g, '\\"') + '"';
         }
+        function quote_template() {
+            if (!options.ascii_only) {
+                str = str.replace(/\\(r|n|u2028|u2029)/g, function(s, c) {
+                    switch(c) {
+                        case "n": return "\n";
+                        case "r": return "\r";
+                        case "u2028": return "\u2028";
+                        case "u2029": return "\u2029";
+                    }
+                    return "\\" + c;
+                });
+            }
+            return '`' + str.replace(/`/g, '\\`') + '`';
+        }
         if (options.ascii_only) str = to_ascii(str);
         switch (options.quote_style) {
           case 1:
+            if (quote == "`" && sq > 0)
+                return quote_template();
```

Comment:
I should have been more specific. My real question was whether back quoted strings containing `${expr}` could reach this particular code in `make_string(str, quote)`. The answer is no. The `${expr}` was already parsed away into segments well before this code is executed, and as such `${expr}` will never appear in `str` when quote is ```"`"```. Maybe a comment to that effect should be put in the code as it wasn't obvious.